R/trackhub.R

```diff
@@ -346,26 +346,54 @@ TrackHubGenome <- function(trackhub, genome, create = FALSE, genomeRecord = Geno
 ###
 
 setClass("TrackContainer",
+         representation("SimpleList"),
+         prototype(elementType = "Track_OR_TrackContainer")
+)
+
+TrackContainer <- function() {
+    new("TrackContainer")
+}
+
+setMethod("[[", "TrackContainer", function(x, i, ...) {
+    getListElement(x, i, ...)
+})
+
+setReplaceMethod("[[", "TrackContainer", function(x, i, ..., value) {
+    setListElement(x, i, value)
+})
+
+setClass("Track",
          representation(
-                        trackList = "list",
-                        childrenList = "list",
-                        parentLookUp = "list"
-                        ),
-         prototype = list(
-                          trackList = list(),
-                          childrenList = list(),
-                          parentLookUp = list()
-         )
+                        track = "character",
+                        bigDataUrl = "character",
+                        children = "TrackContainer"
+                        )
 )
 
-getTabCountList <- function(contentdf) {
-    tabCountList <- vapply(contentdf$V1, function(x) {
-        tmp <- gregexpr("^(\\t)+", x)
-        tabCount <- attr(tmp[[1]], "match.length")
-    }, numeric(1), USE.NAMES = FALSE)
-    tabCountList
+setMethod("[", "Track", function(x, i, ...) {
+    x@children
+})
+
+setReplaceMethod("[", "Track", function(x, i, ..., value) {
+    x@children <- value
+})
+
+setMethod("[[", "Track", function(x, i, ...) {
+    if (i <= length(x@children))
```

Comment:
Should be no need for this condition.  We want `getListElement()` to fail with a "subscript out of bounds error" for consistency with ordinary list behavior. This would return `NULL` instead.